### ISSUE TITLE
Support for mining with multiple accounts

### ISSUE THREAD

**Issue by moligod** [DevilXD/TwitchDropsMiner#31]
Currently, only one account can be added
![image](https://user-images.githubusercontent.com/89651085/179290398-f7ffc60f-a185-4ede-aad4-74d32ae1df8a.png)

**Comment by DevilXD** [comment]
Hello o/

This has already been requested as a feature in: #22 - for now, it's a "maybe in the future" from my side. You can leave this issue open so that everyone else can see it's already been requested, but otherwise you can check out the linked issue for a "workaround" that'd allow you to use the miner with multiple accounts.

**Comment by iFireMonkey** [comment]
So an interesting thing about the work around, for some reason whenever I have two instances of the program running it causes games to completely break, making it basically impossible to play a game with two programs open.

For Overwatch it causes the game to just be a black screen while other games like Fortnite the game just doesn't launch or launches without mouse detection.

The only arguments being used are `--no-run-check` and `--tray` however i've tried with just no run check and had the same issue.

**Comment by DevilXD** [comment]
Hmm. This is actually a potential collision in play here, as the tray icon is registered and launched as a separate application window, but under the same name. This may cause two or more instances to "criss-cross" themselves on the tray icon access, which ends up causing issues with Windows itself, somehow.

Does this still happen if you just have two application windows open, but not put into the tray?

**Comment by iFireMonkey** [comment]
Hey! Just went and tested this, after launching both copies under just `--no-run-check` it does appear that the issue still for whatever reason occurs.

I'm currently on Windows 11 Version 22H2 (Build 22623.870) and with only one open everything is fine however the moment I open the second one things such as video games start breaking and becoming unusable (mainly games from my experience), and once I close one of the two running copies everything starts working again like normal.

**Comment by DevilXD** [comment]
That's really strange. I imagine it can be related to me using Python 3.8 to build this application, it's quite old by now, and Win11 is pretty new. You could try rebuilding the application with 3.10 to see if it'd help eliminate the issue. Please follow the instructions from here: https://github.com/DevilXD/TwitchDropsMiner#advanced-usage--build-instructions

For the Python installation, you can use the #1st point from here: https://github.com/DevilXD/TwitchDropsMiner/issues/12#issuecomment-1087831162  I recommend using the "All users" install with `C:/Python310` as the installation path, to avoid issues.

**Comment by iFireMonkey** [comment]
Hey! I've gone ahead and built the program using Python 3.10, sadly it appears this issue still occurs, I also tested closing one of the two running instances and just like the 3.8 build the issue stops.

Ironically enough however, by launching one instance VIA the `pythonw main.py` way and having one open through a normal .exe this issue doesn't occur at all

**Comment by DevilXD** [comment]
I have no idea why this may be happening. Other than the tray icon, there's no operating system interaction beyond the GUI that's displayed. The `aiohttp` library handles all socket-related operations and there should be no problems with more than one instance of the library running at the same time either.

Unfortunately, I think I'm unable to help you with this issue. It sounds like something Windows 11 related specifically, which given it's current scarcity (I'm still on Windows 7, Win10 at work) I'm unable to even test it myself.

**Comment by lmde000111** [comment]
Is there any way to open multiple programs or log in to multiple accounts?

**Comment by Walter-o** [comment]
I have had success with running multiple instances of this on linux (rpi3b+) while running python 3.11 with the source code.

My setup was:
xvfb - For virtual display (i only have 1gb ram on rpi3b+) 
x11vnc - For viewing that virtual display
systemd - For auto-starting the virtual display stuff and drops miners
pyenv - To switch between python versions without installing them into system (i used python 3.11 for the drops miner)
pyenv-virtualenv - Its not possible to use venv in pyenv so i had to use this

I was running 3 accounts, they ran ok, and i didn't get any rate limiting errors.
Also i ran each account in a seperate folder with a copy of the source code to avoid conflicts.

**Comment by headpunch** [comment]
I manage to run 11 / 12 in same time with `--no-run-check` but after I was experiencing weird things, when I tried to open next one, I didn't show up, I couldn't open any folder.

**Comment by axi92** [comment]
What are your thoughts on handling multiple accounts with docker? @DevilXD 
That way the app does not need to support that and for the most usage its one time login/config and then just starting it or keep it running for a time.

With a docker image it could be configured once without docker and then the config is mounted into the container. So a compose file could start farming for 3 accounts or more with one cmd.

The obvious bad thing with this may be is that its more easy for users to farm with more accounts and Twitch starts to do something against this.

**Comment by DevilXD** [comment]
Yeah, I'd like to not use docker, or any form of deployment, really. I also still don't really see the need for having to support more than one account, so this issue just... hangs indefinitely. What are you all doing if you need/have more than one Twitch account for your games? I still just can't understand that. Nobody here has had any convincing arguments for this yet. I'm running just one account and that's always been enough for me. Even per service, I never really found a reason to have more than one account elsewhere, that I'd use in a similar way to the main account. I do have a 2nd account on Reddit for example, but that's specifically for a bot, so that it responds under it's own account, instead of using mine.

**Comment by rapcrown** [comment]
> Evet, docker veya herhangi bir dağıtım biçimi kullanmak istemiyorum. Ayrıca hala birden fazla hesabı desteklemeye gerek olduğunu gerçekten göremiyorum, bu yüzden bu sorun sadece... süresiz olarak askıda kalıyor. Oyunlarınız için birden fazla Twitch hesabına ihtiyacınız varsa/varsa ne yapıyorsunuz? Bunu hala anlayamıyorum. Burada henüz kimse bunun için ikna edici bir argüman sunmadı. Sadece bir hesap çalıştırıyorum ve bu benim için her zaman yeterli oldu. Hizmet başına bile, ana hesaba benzer şekilde kullanacağım birden fazla hesaba sahip olmak için başka bir yerde gerçekten bir neden bulamadım. Örneğin Reddit'te 2. bir hesabım var, ancak bu özellikle bir bot için, böylece benim hesabımı kullanmak yerine kendi hesabı altında yanıt veriyor.


We have multiple accounts in the game, and most people want to use them to get drops.

**Comment by xenxgod666** [comment]
When can you add support for multiple accounts?

**Comment by DevilXD** [comment]
There are still no plans to support more than a single account by the miner. #220 is the closest that I can get to even thinking about doing that, and even then, I still don't see a good reason for it. If you really need to run the miner on more than one account, just run multiple instances of it.

**Comment by FomaDima** [comment]
"When I tried running several program windows at once, my computer started lagging badly. I couldn't open the browser, or even File Explorer [the modern name for 'My Computer'] for that matter."

**Comment by DevilXD** [comment]
@FomaDima I said running multiple instances is possible, not that you won't run into problems with it. It's not exactly supported either. It's possible to run into the rate limits with just 2 instances already, and any number higher than that will only bring you more and more trouble to it. The rate limit is a Twitch limitation, so there's little one can do about it.

It the high CPU usage issue happens for a single instance, I can try taking a look at it. Otherwise, as I said, multiple instances is possible, but not supported.

**Comment by FomaDima** [comment]
Excuse me. I just started Googling, I got curious. I wanted to know if this is the same for everyone or just me: I came across a topic and decided to write about it. It's just funny that my browser isn't working, even though I can launch and play a couple of 'heavy' games

"Yeah, I was already looking through topics and saw you guys aren't planning that kind of support anyway. Never mind, don't let me distract you, sorry."
